group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -712,13 +712,14 @@ public void run() {
                         try {
                             // Apply the records to the state machine.
                             if (result.replayRecords()) {
-                                result.records().forEach(record ->
+                                for (int i = 0; i < result.records().size(); i++) {
                                     context.coordinator.replay(
+                                        prevLastWrittenOffset + i,
```

Comment:
is this the only way to get the offsets? 😅

Reply:
yes. the only way to speculate about what the offset will be in the log based on the last written offset. as the coordinator is the single writer to the log, it works.

Reply:
could we maybe leave a comment about that?

Reply:
Sure.